lib/matplotlib/axes/_base.py

```diff
@@ -3174,15 +3174,14 @@ def set_axisbelow(self, b):
         --------
         get_axisbelow
         """
+        # Check that b is True, False or "line"
         self._axisbelow = axisbelow = validate_axisbelow(b)
         if axisbelow is True:
             zorder = 0.5
         elif axisbelow is False:
             zorder = 2.5
         elif axisbelow == "line":
             zorder = 1.5
-        else:
-            raise ValueError("Unexpected axisbelow value")
```

Comment:
This sort of shows why adding tests for error messages can be worthwhile. Here it turns out that this cannot be reached anymore since `validate_axisbelow` will error for any other options.

(So I should probably change the last `elif` to `else`.)

Reply:
Thanks! Good idea! I applied it in a few other cases as well and for others used else instead of elif when the arguments had already been checked.